tests/regression/roles/sonic_static_routes/defaults/main.yml

```diff
@@ -0,0 +1,181 @@
+          - prefix: '7.0.0.0/8'
+            next_hops:
+              - index:
+                  nexthop_vrf: '{{vrf_1}}'
+                  next_hop: '3.0.0.0'
+  - name: test_case_02
+    description: Modify static routes configuration
+    state: merged
+    input:
+      - vrf_name: 'default'
+        static_list:
+         - prefix: '2.0.0.0/8'
+           next_hops:
+             - index:
+                 interface: 'Ethernet4'
+               metric: 10
+               tag: 20
+               track: 30
+             - index:
+                next_hop: '3.0.0.0'
+               metric: 8
+               tag: 10
+               track: 12
+      - vrf_name: '{{vrf_1}}'
+        static_list:
+          - prefix: '3.0.0.0/8'
+            next_hops:
+              - index:
+                  blackhole: True
+                  interface: 'Ethernet4'
+                  nexthop_vrf: '{{vrf_2}}'
+                  next_hop: '5.0.0.0'
+                metric: 11
+                tag: 22
+                track: 33
+      - vrf_name: '{{vrf_2}}'
+        static_list:
+          - prefix: '1.0.0.0/8'
+            next_hops:
+              - index:
+                  interface: 'Ethernet6'
+                  nexthop_vrf: '{{vrf_1}}'
+                metric: 6
+                tag: 7
+                track: 8
+          - prefix: '7.0.0.0/8'
+            next_hops:
+              - index:
+                  nexthop_vrf: '{{vrf_1}}'
+                  next_hop: '3.0.0.0'
+                metric: 9
+                tag: 10
+                track: 11
+  - name: test_case_03
+    description: Delete static routes metric, tag, and track configuration
+    state: deleted
+    input:
+      - vrf_name: 'default'
+        static_list:
+         - prefix: '2.0.0.0/8'
+           next_hops:
+             - index:
+                 interface: 'Ethernet4'
+               metric: 10
+               tag: 20
+               track: 30
+             - index:
+                next_hop: '3.0.0.0'
+               metric: 8
+               tag: 10
+               track: 12
+      - vrf_name: '{{vrf_1}}'
+        static_list:
+          - prefix: '3.0.0.0/8'
+            next_hops:
+              - index:
+                  blackhole: True
+                  interface: 'Ethernet4'
+                  nexthop_vrf: '{{vrf_2}}'
+                  next_hop: '5.0.0.0'
+                metric: 11
+                tag: 22
+                track: 33
+              - index:
+                  interface: 'eth0'
+                  nexthop_vrf: '{{vrf_2}}'
+                  next_hop: '4.0.0.0'
+  - name: test_case_04
+    description: Delete static route index configuration
+    state: deleted
+    input:
+      - vrf_name: 'default'
+        static_list:
+         - prefix: '2.0.0.0/8'
+           next_hops:
+             - index:
+                 interface: 'Ethernet4'
+             - index:
+                next_hop: '3.0.0.0'
+      - vrf_name: '{{vrf_1}}'
```

Comment:
Deletion of the only remaining next hop for the only remaining prefix in vrf_1 leaves vrf_1 with no prefixes (no routes). As a result, the only deletion at the VRF level (in test_case_05, for vrf 1) is a no-op and doesn't actually test the case of deletion at the vrf level. Instead, since this test is already doing a good job of verifying handling for next hop deletion, it would be better to take this block (next hop deletion for vrf_1) out of test_case_04. This would allow test_case_05 to verify the functionality of vrf level deletion.